identity-issuance/src/main/java/com/android/identity/issuance/wallet/ApplicationSupportState.kt

```diff
@@ -0,0 +1,149 @@
+import kotlinx.io.bytestring.ByteString
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.buildJsonObject
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+@FlowState(flowInterface = ApplicationSupport::class)
+@CborSerializable
+class ApplicationSupportState(
+    var clientId: String
+) {
+    companion object {
+        const val URL_PREFIX = "landing/"
+        const val TAG = "ApplicationSupportState"
+    }
+
+    @FlowMethod
+    suspend fun createLandingUrl(env: FlowEnvironment): String {
+        val storage = env.getInterface(Storage::class)!!
+        val id = storage.insert("Landing", "", ByteString(LandingRecord(clientId).toCbor()))
+        Logger.i(TAG, "Created landing URL '$id'")
+        return URL_PREFIX + id
+    }
+
+    @FlowMethod
+    suspend fun getLandingUrlStatus(env: FlowEnvironment, baseUrl: String): String? {
+        if (!baseUrl.startsWith(URL_PREFIX)) {
+            throw IllegalStateException("baseUrl must start with $URL_PREFIX")
+        }
+        val storage = env.getInterface(Storage::class)!!
+        val id = baseUrl.substring(URL_PREFIX.length)
+        Logger.i(TAG, "Querying landing URL '$id'")
+        val recordData = storage.get("Landing", "", id) ?:
+            throw LandingUrlUnknownException("No landing url '$id'")
+        val record = LandingRecord.fromCbor(recordData.toByteArray())
+        if (record.resolved != null) {
+            Logger.i(TAG, "Removed landing URL '$id'")
+            storage.delete("Landing", "", id)
+        }
+        return record.resolved
+    }
+
+    @FlowMethod
+    fun createJwtClientAssertion(
+        env: FlowEnvironment, attestation: KeyAttestation, targetIssuanceUrl: String): String {
+        val settings = WalletServerSettings(env.getInterface(Configuration::class)!!)
+
+        validateKeyAttestation(
+            attestation.certChain!!,
+            null,  // no challenge check
+            settings.androidRequireGmsAttestation,
+            settings.androidRequireVerifiedBootGreen,
+            settings.androidRequireAppSignatureCertificateDigests
+        )
+
+        check(attestation.certChain!!.certificates[0].ecPublicKey == attestation.publicKey)
+        return createJwtClientAssertion(env, attestation.publicKey, targetIssuanceUrl)
+    }
+
+    // Not exposed as RPC!
+    fun createJwtClientAssertion(
+        env: FlowEnvironment,
+        clientPublicKey: EcPublicKey,
+        targetIssuanceUrl: String
+    ): String {
+        val attestationData = env.cache(
+            ClientAttestationData::class,
+            targetIssuanceUrl
+        ) { configuration, resources ->
+            // These are basically our credentials to talk to a particular OpenID4VCI issuance
+            // server. So in real life this may need to be parameterized by targetIssuanceUrl
+            // if we support many issuance servers.
```

Comment:
We support Funke and our own and this code is sufficient for that. This would also work for other non-production cases. If we want to support more realistic production-grade setups, we would need to define how exactly the trust between clients and servers achieved. I am not really sure if this is in scope.